backends/hip-ref/ceed-hip-ref-basis.c

```diff
@@ -35,20 +35,12 @@ static int CeedBasisApplyCore_Hip(CeedBasis basis, bool apply_add, const CeedInt
   if (u != CEED_VECTOR_NONE) CeedCallBackend(CeedVectorGetArrayRead(u, CEED_MEM_DEVICE, &d_u));
   else CeedCheck(eval_mode == CEED_EVAL_WEIGHT, ceed, CEED_ERROR_BACKEND, "An input vector is required for this CeedEvalMode");
   if (apply_add) CeedCallBackend(CeedVectorGetArray(v, CEED_MEM_DEVICE, &d_v));
-  else CeedCallBackend(CeedVectorGetArrayWrite(v, CEED_MEM_DEVICE, &d_v));
-
-  // Clear v for transpose operation
-  if (is_transpose && !apply_add) {
-    CeedInt  num_comp, q_comp, num_nodes, num_qpts;
-    CeedSize length;
-
-    CeedCallBackend(CeedBasisGetNumComponents(basis, &num_comp));
-    CeedCallBackend(CeedBasisGetNumQuadratureComponents(basis, eval_mode, &q_comp));
-    CeedCallBackend(CeedBasisGetNumNodes(basis, &num_nodes));
-    CeedCallBackend(CeedBasisGetNumQuadraturePoints(basis, &num_qpts));
-    length = (CeedSize)num_elem * (CeedSize)num_comp * (t_mode == CEED_TRANSPOSE ? (CeedSize)num_nodes : ((CeedSize)num_qpts * (CeedSize)q_comp));
-    CeedCallHip(ceed, hipMemset(d_v, 0, length * sizeof(CeedScalar)));
+  else {
```

Comment:
I lost the comment here, but the style is to use {} for both clauses if one of them needs it
```